include/ccmath/math/expo/log10.hpp

```diff
@@ -10,24 +10,23 @@
 
 #pragma once
 
+#include "ccmath/internal/math/generic/builtins/expo/log10.hpp"
+
 #include <type_traits>
 
 namespace ccm
 {
 	template <typename T, std::enable_if_t<!std::is_integral_v<T>, bool> = true>
 	constexpr T log10([[maybe_unused]] T num)
 	{
-		#if defined(__GNUC__) && (__GNUC__ > 6 || (__GNUC__ == 6 && __GNUC_MINOR__ >= 1)) && !defined(__clang__)
-		if constexpr (std::is_same_v<T, float>) { return __builtin_log10f(num); }
-		if constexpr (std::is_same_v<T, double>) { return __builtin_log10(num); }
-		if constexpr (std::is_same_v<T, long double>) { return __builtin_log10l(num); }
-		return static_cast<T>(__builtin_expm1l(static_cast<long double>(num)));
-		#else
-		if constexpr (std::is_same_v<T, float>) { return 0; }
-		if constexpr (std::is_same_v<T, double>) { return 0; }
-		if constexpr (std::is_same_v<T, long double>) { return 0; }
-		return 0;
-		#endif
+		if constexpr (ccm::builtin::has_constexpr_log10<T>) { return ccm::builtin::log10(num); }
+		else
+		{
+			if constexpr (std::is_same_v<T, float>) { return 0; }
+			if constexpr (std::is_same_v<T, double>) { return 0; }
+			if constexpr (std::is_same_v<T, long double>) { return 0; }
+			return 0;
```

Comment:
And here, shouldn't it be T{0} or static_cast<T>(0); ?

Reply:
For now this is a placeholder as the code has not been implemented yet. I'd say the details of how it returns really does not matter at this moment.

Reply:
I understand